acceptance/tests/failover/timeout_failover.rb

```diff
@@ -0,0 +1,44 @@
+require 'pxp-agent/config_helper.rb'
+require 'pxp-agent/test_helper.rb'
+
+test_name 'C97964 - agent should use next broker if primary is timing out' do
+
+  blocked_broker_instance = 0
+  teardown do
+    unblock_pcp_broker(master,blocked_broker_instance)
+    broker_instance = 0 # 0 indexed
+    kill_pcp_broker(master)
+    run_pcp_broker(master, broker_instance)
+  end
+
+  step 'Ensure each agent host has pxp-agent configured with multiple uris, running and associated' do
+    agents.each do |agent|
+      on agent, puppet('resource service pxp-agent ensure=stopped')
+      num_brokers = 2
+      pxp_config = pxp_config_hash_using_puppet_certs(master, agent, num_brokers)
+      pxp_config['allowed-keepalive-timeouts'] = 0
+      create_remote_file(agent, pxp_agent_config_file(agent), pxp_config.to_json.to_s)
+      on(agent, "rm -rf #{logfile(agent)}")
+      on agent, puppet('resource service pxp-agent ensure=running')
+      show_pcp_logs_on_failure do
+        assert(is_associated?(master, "pcp://#{agent}/agent"),
```

Comment:
There are two brokers in play now; what exactly is being asserted here? From reading the recent changes to the helper code; I can see that this call to is_associated? will go and use broker 0; but I think the test could be more readable and state that it is asserting association on the primary broker.

It might also be diligent to assert that the agent is not associated on the replica broker. I don't currently know any possibility how it might report as associated on both; but if we ever got into that weird state, it would be nicer to fail the test now rather than get a false result